src/pokemon_summary_screen.c

```diff
@@ -0,0 +1,3680 @@
+void sub_81C4C84(u8 a);
+void sub_81C13B0(u8 a, u8 b);
+void sub_81C14BC(struct Pokemon *mon, u8 a, u8 b);
+void sub_81C15EC(struct BoxPokemon *mon, u8 a, u8 b);
+void sub_81C40A0(u8 a, u8 b);
+void sub_81C4568(u8 a, u8 b);
+void sub_81C174C(u8 taskId);
+u8 sub_81C18A8();
+void sub_81C18F4(u8 a);
+u8 sub_81B6D14(u16 a);
+void sub_81C1940(u8 taskId);
+void sub_81C4154();
+void sub_81C1F80(u8 taskId);
+void sub_81C2628();
+void sub_81C2794();
+void sub_81C27DC(struct Pokemon *mon, u16 a);
+void sub_81C2D68(u8 a);
+void sub_81C335C();
+void sub_81C33CC();
+void sub_81C3428();
+void sub_81C349C();
+void sub_81C2EC4();
+void sub_81C2F5C();
+void sub_81C2FD8();
+void sub_81C302C();
+void sub_81C307C();
+void sub_81C3194();
+void sub_81C31C0();
+u8 sub_81C3304();
+void sub_81C31F0(u8 *a);
+u8 sub_81C3220();
+u8 sub_81C32BC();
+void sub_81C35E4();
+void sub_81C3690();
+void sub_81C3710();
+void sub_81C37D8();
+void sub_81C3808();
+void sub_81C3890();
+void sub_81C38C0();
+void sub_81C3B08(u8 a);
+void sub_81C3E2C(u8 a);
+
+void SetBgAttribute(u8 bg, u8 attributeId, u8 value);
+
+bool8 sub_81BFB10();
+u8 sub_81B1250();
+
+union unkUnion
+{
+    struct Pokemon mon[6];
+    struct BoxPokemon boxMon[6];
+};
+
+
+u8 sub_80D214C(union unkUnion* a, u8 b, u8 c, u8 d);
+
+struct PokeSummary
+{
+    u16 species; // 0x0
+    u16 species2; // 0x2
+    u8 isEgg; // 0x4
+    u8 level; // 0x5
+    u8 ribbons; // 0x6
+    u8 unk7; // 0x7
+    u8 altAbility; // 0x8
+    u8 metLocation; // 0x9
+    u8 metLevel; // 0xA
+    u8 metGame; // 0xB
+    u32 pid; // 0xC
+    u32 exp; // 0x10
+    u16 moves[4]; // 0x14
+    u8 pp[4]; // 0x1C
+    u16 currentHP; // 0x20
+    u16 maxHP; // 0x22
+    u16 atk; // 0x24
+    u16 def; // 0x26
+    u16 spatk; // 0x28
+    u16 spdef; // 0x2A
+    u16 speed; // 0x2C
+    u16 item; // 0x2E
+    u16 friendship; // 0x30
+    u8 OTGender; // 0x32
+    u8 nature; // 0x33
+    u8 ppBonuses; // 0x34
+    u8 sanity; // 0x35
+    u8 OTName[8]; // 0x36
+    u8 unk3E[9]; // 0x3E
+    u32 OTID; // 0x48
+};
+
+struct UnkSummaryStruct
+{
+    /*0x00*/ union unkUnion *unk0;
+    /*0x04*/ void *unk4;
+    /*0x08*/ void *unk8;
+    /*0x0C*/ struct Pokemon currentPoke;
+    /*0x70*/ struct PokeSummary summary;
+    u16 unkTilemap0[0x400];
+    u16 unkTilemap0_1[0x400];
+    u16 unkTilemap1[0x400];
+    u16 unkTilemap1_1[0x400];
+    u16 unkTilemap2[0x400];
+    u16 unkTilemap2_1[0x400];
+    u16 unkTilemap3[0x400];
+    u16 unkTilemap3_1[0x400];
+    u8 unk40BC;
+    u8 unk40BD;
+    u8 unk40BE;
+    u8 unk40BF;
+    u8 unk40C0;
+    u8 unk40C1;
+    u8 unk40C2;
+    u8 unk40C3;
+    u16 unk40C4;
+    u8 unk40C6;
+    u8 unk40C7;
+    u8 unk40C8;
+    u8 unk40C9;
+    u8 unk40CA;
+    u8 unk40CB[8];
+    u8 unk40D3;
+    u8 unk40D4;
+    u8 unk40D5;
+    u8 unk_filler5[0x19];
+    u8 unk40EF;
+    s16 unk40F0;
+    u8 unk_filler4[6];
+};
+
+void sub_81BF8EC(u8 a, void *b, u8 c, u8 d, void *e)
+{
+    u8 byte;
+    gUnknown_0203CF1C = AllocZeroed(0x40F8);
```

Comment:
Could this size be replaced with sizeof(UnkSummaryStruct)?

Reply:
nope, tried to